services/metrics-api/src/helpers/metricBuilder.ts

```diff
@@ -0,0 +1,49 @@
+import { isMetricValueWithMeta } from './metrics';
+import type { MaybeMetricMeta, Metric, MetricType, MetricValue } from './metrics.types';
+
+export default class MetricBuilder<TMeta extends MaybeMetricMeta = void> {
+  #metric: Metric<TMeta>;
+
+  constructor(name: string) {
+    this.#metric = {
+      name,
+      values: [],
+    };
+  }
+
+  setHelp(help: string): MetricBuilder<TMeta> {
+    this.#metric.help = help;
+    return this;
+  }
+
+  setType(type: MetricType): MetricBuilder<TMeta> {
```

Comment:
Kan detta skrivas som? Blir kanske lite mer oopigt?
```typescript
set help() {}
set type() {}
get metric() {}
```

Reply:
Man kan tyvärr inte returnera `this` och kedja då verkar det som, och då kan vi inte längre applicera Builder-pattern.